command/executor_plugin.go

```diff
@@ -69,3 +84,75 @@ func (e *ExecutorPluginCommand) Run(args []string) int {
 	})
 	return 0
 }
+
+type circbufWriter struct {
+	buf     *circbuf.Buffer
+	err     error
+	bufLock sync.Mutex
+	wr      io.Writer
+	flushCh chan struct{}
+}
+
+func newCircbufWriter(w io.Writer) *circbufWriter {
+	buf, _ := circbuf.NewBuffer(circleBufferSize)
+	c := &circbufWriter{
+		buf:     buf,
+		wr:      w,
+		flushCh: make(chan struct{}),
+	}
+	go c.flushLoop()
+	return c
+}
+
+func (c *circbufWriter) Write(p []byte) (nn int, err error) {
+	if c.err != nil {
```

Comment:
Since `c.err` is set within the lock you must do the read inside the lock as well.